lib/Doctrine/DBAL/Connection.php

```diff
@@ -1589,11 +1589,17 @@ public function ping()
             return $this->_conn->ping();
         }
 
+        set_error_handler(function ($severity, $message) {
+            throw new \PDOException($message, $severity);
```

Comment:
You're right, it should throw an exception, but it's triggering a warning. Anyway, Doctrine should contain a fix for the driver.

I was thinking of implementing it directly in the `Doctrine\DBAL\Driver\PDOConnection::query()`, but changing error handler for every query wouldn't be very fast.